pymeasure/instruments/easttester/et35xx.py

```diff
@@ -0,0 +1,307 @@
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.generic_types import SCPIMixin
+from pymeasure.instruments.validators import strict_range, strict_discrete_set
+
+import enum
+
+
+class BiasSource(enum.Enum):
+    """Supported bias voltage sources."""
+    INTERNAL = enum.auto()
+    EXTERNAL = enum.auto()
+
+
+class MeasurementRange(enum.Enum):
+    """Supported measurement ranges."""
+    RANGE_30 = enum.auto()
+    RANGE_100 = enum.auto()
+    RANGE_300 = enum.auto()
+    RANGE_1k = enum.auto()
+    RANGE_3k = enum.auto()
+    # RANGE_10k = enum.auto()
+    RANGE_30k = enum.auto()
+    # RANGE_100k = enum.auto()
+    RANGE_300k = enum.auto()
+    RANGE_1M = enum.auto()
+
+
+class MeasurementSpeed(enum.Enum):
+    """Supported measurement speeds."""
+    FAST = enum.auto()
+    MEDIUM = enum.auto()
+    SLOW = enum.auto()
+    CUSTOM = enum.auto()
+
+
+class MeasurementType(enum.Enum):
+    """Supported measurement types."""
+    CpD = enum.auto()
+    CpQ = enum.auto()
+    CpG = enum.auto()
+    CpRp = enum.auto()
+    CsD = enum.auto()
+    CsQ = enum.auto()
+    CsRs = enum.auto()
+    LpD = enum.auto()
+    LpQ = enum.auto()
+    LpG = enum.auto()
+    LpRp = enum.auto()
+    LsD = enum.auto()
+    LsQ = enum.auto()
+    LsRs = enum.auto()
+    RX = enum.auto()
+    ZDeg = enum.auto()
+    ZRad = enum.auto()
+    GB = enum.auto()
+    YDeg = enum.auto()
+    YRad = enum.auto()
+
+
+class TriggerSource(enum.Enum):
+    """Supported trigger sources."""
+    INTERNAL = enum.auto()
+    MANUAL = enum.auto()
+    EXTERNAL = enum.auto()
+    BUS = enum.auto()
+
+
+class ET3510(SCPIMixin, Instrument):
+    """Control an East Tester ET3510 LCR meter."""
+
+    def __init__(self, adapter, name="East Tester ET3510", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            read_termination='\n',
+            **kwargs
+        )
+
+    BiasSource = BiasSource
+    MeasurementRange = MeasurementRange
+    MeasurementSpeed = MeasurementSpeed
+    MeasurementType = MeasurementType
+    TriggerSource = TriggerSource
+
+    BOOL_TO_ON_OFF = {True: "ON", False: "OFF"}
+
+    #
+    # AMPLitude subsystem
+    #
+    automatic_level_control = Instrument.control(
+        "AMPL:ALC?", "AMPL:ALC %s",
+        """Get or set automatic level control.""",
```

Comment:
We use the word "Control" to indicate "Get or set":
```suggestion
        """Control automatic level control.""",
```